### ISSUE TITLE
Express Vite template loads white screen in production mode

### ISSUE THREAD

**Issue by numso** [remix-run/remix#9275]
### Reproduction

- npx create-remix@latest --template remix-run/remix/templates/express
- npm run build
- npm start

(note: npm run dev works fine. seems to be an issue with the production build)

### System Info

```shell
System:
    OS: macOS 13.1
  Binaries:
    Node: 20.7.0 - ~/.nvm/versions/node/v20.7.0/bin/node
    npm: 10.1.0 - ~/.nvm/versions/node/v20.7.0/bin/npm
  Browsers:
    Arc: 1.39.0
  npmPackages:
    @remix-run/dev: ^2.8.1 => 2.8.1 
    @remix-run/express: ^2.8.1 => 2.8.1 
    @remix-run/node: ^2.8.1 => 2.8.1 
    @remix-run/react: ^2.8.1 => 2.8.1 
    vite: ^5.1.0 => 5.2.10
```

### Used Package Manager

npm

### Expected Behavior

I expected to see the default "Welcome to Remix" page.

### Actual Behavior

The "Welcome to Remix" page flashes on the screen and then the screen goes completely blank. These errors are thrown in the console. I did this in a private window with no extensions.

<img width="875" alt="image" src="https://github.com/remix-run/remix/assets/1761757/92ced539-f0c8-459a-bf60-ee5e0fd3696c">

**Comment by hugocbp** [comment]
Same thing happens with the Cloudfare template https://test-remix-cloudfare-app.pages.dev/

A fresh no-template Remix install is also failing on Fly.io.

**Comment by huijiewei** [comment]
override rollup to 4.15.0, this is rollup issue.

**Comment by hugocbp** [comment]
> override rollup to 4.15.0, this is rollup issue.

Thank you! That fixed it on both cases.

**Comment by changeyoucant** [comment]
> override rollup to 4.15.0, this is rollup issue.

Hello, have a nice day ! I'm encountering the same issue. The page is displaying now, but the error persists. I'm not sure if I should ignore this error.

**Comment by AdiRishi** [comment]
@changeyoucant are you sure you've set the override for rollup correctly? That error disappeared for me when I fixed rollup as mentioned above.
Add this to your `package.json` (assuming you're using `pnpm`)
```json
"pnpm": {
    "overrides": {
      "rollup": "4.15.0"
    }
}
```

**Comment by brophdawg11** [comment]
As folks have pointed out above, this is due to an issue in `rollup@4.16` which is being investigated and should hopefully be fixed/reverted soon: https://github.com/rollup/rollup/pull/5443

Also a duplicate of https://github.com/remix-run/remix/issues/9271

The proper workaround for now is to use `rollup@4.15` either via direct install or by adding to the `overrides` section of your `package.json`.

**Comment by brophdawg11** [comment]
This should b resolved in rollup 4.16.2: <https://github.com/rollup/rollup/releases/tag/v4.16.2>

**Comment by Cmoen11** [comment]
> This should be resolved in rollup 4.16.2: https://github.com/rollup/rollup/releases/tag/v4.16.2

Forgive me if I'm wrong.. But since this affects production builds, it should be announced some where? As for me, when i look up issues - i search for open issues - and this is now closed. So it might be hard for someone to figure out...

**Comment by brophdawg11** [comment]
Why wouldn't this be closed if the issue is now fixed in rollup `4.16.2`?

**Comment by Cmoen11** [comment]
Why? Because when you pull down the template and try to deploy it you be left out not knowing why your pages aren't rendering...? And by logic you will look for opened issues, not closed issues to try to figure out why...

It is not obvious for non contributors that it is rollup that is the issue
